In [5]:
from transformers import AutoTokenizer

from datasets import load_dataset
import pandas as pd


In [6]:
checkpoint = "google/flan-t5-base"
dataset_name = "xsum"

ds = load_dataset(dataset_name)
tokenizer = AutoTokenizer.from_pretrained(checkpoint)

  0%|          | 0/3 [00:00<?, ?it/s]

In [7]:
example = ds["train"][0]
example

{'document': 'The full cost of damage in Newton Stewart, one of the areas worst affected, is still being assessed.\nRepair work is ongoing in Hawick and many roads in Peeblesshire remain badly affected by standing water.\nTrains on the west coast mainline face disruption due to damage at the Lamington Viaduct.\nMany businesses and householders were affected by flooding in Newton Stewart after the River Cree overflowed into the town.\nFirst Minister Nicola Sturgeon visited the area to inspect the damage.\nThe waters breached a retaining wall, flooding many commercial properties on Victoria Street - the main shopping thoroughfare.\nJeanette Tate, who owns the Cinnamon Cafe which was badly affected, said she could not fault the multi-agency response once the flood hit.\nHowever, she said more preventative work could have been carried out to ensure the retaining wall did not fail.\n"It is difficult but I do think there is so much publicity for Dumfries and the Nith - and I totally apprecia

In [8]:
tk_doc = tokenizer(ds["train"]["document"])["input_ids"]
tk_summary = tokenizer(ds["train"]["summary"])["input_ids"]
pd.set_option("display.float_format", lambda x: "%.1f" % x)

df = pd.DataFrame(
    {"document": [len(d) for d in tk_doc], "summary": [len(s) for s in tk_summary]}
)
print(df.describe())

       document  summary
count  204045.0 204045.0
mean      523.9     30.4
std       438.2      8.3
min         1.0      3.0
25%       247.0     25.0
50%       410.0     30.0
75%       680.0     35.0
max     39488.0    178.0


- `39488`... You are the one to blame. Definitely need truncation to deal with such outlier. Remember to look at raw data and anticipate outliers next time. 
- 75% = 680, `max_length=1024` should do the job without sacrificing too much input info. 